sklearn/discriminant_analysis.py

```diff
@@ -254,6 +255,18 @@ def __init__(self, solver='svd', shrinkage=None, priors=None,
         self.store_covariance = store_covariance  # used only in svd solver
         self.tol = tol  # used only in svd solver
 
+    @property
+    def covariance_(self):
+        if hasattr(self, '_covariance_') and not self.store_covariance:
```

Comment:
`_solve_svd` seems to already store `covariance_` iff `store_covariance` is set, so the warning serves no purpose. Basically, this PR should just drop back to dealing with the inconsistency between covariance/covariances.